PokeAlarm/Utilities/PvpUtils.py

```diff
@@ -59,6 +59,101 @@ def min_level(limit, monster):
+    stardust_table = [200, 400, 600, 800, 1000, 1300, 1600, 1900, 2200,
+                      2500, 3000, 3500, 4000, 4500, 5000, 6000, 7000,
+                      8000, 9000, 10000, 11000, 12000, 13000, 14000, 15000]
+
+    tmp_level = start_level
+    stardust_cost = 0
+    while tmp_level != target_level:
+        tmp_level += 0.5
+        cost_id = int(tmp_level - 1.5) // 2
+        stardust_cost += stardust_table[cost_id]
+
+    return f'{stardust_cost:,}'.replace(',', ' ')
+
+
+def calculate_evolution_cost(monster_id, evolution_id, evolutions):
+    evolution_candy_costs = {
+        12: [10, 13, 16, 265, 293, 519, 664],
+        25: [1, 4, 7, 19, 29, 32, 41, 43, 60, 63, 66, 69, 74, 92, 111, 116,
+             133, 137, 147, 152, 155, 158, 161, 172, 173, 174, 175, 179, 183,
+             187, 220, 236, 238, 239, 240, 246, 252, 255, 258, 270, 273, 280,
+             287, 298, 304, 328, 355, 363, 371, 374, 387, 390, 393, 396, 403,
+             406, 439, 440, 443, 446, 495, 498, 501, 506, 535, 540, 543, 551,
+             574, 577, 582, 599, 602, 607, 610, 633, 650, 653, 656, 661, 669,
+             679, 704, 722, 725, 728, 731, 736, 761, 782, 789, 810, 813, 816,
+             821, 824, 837, 856, 859, 885],
+        50: [11, 14, 17, 21, 23, 25, 27, 35, 37, 39, 46, 48, 50, 52, 54, 56,
+             58, 72, 77, 79, 81, 83, 84, 86, 88, 90, 95, 96, 98, 100, 102,
+             104, 109, 113, 118, 120, 123, 138, 140, 163, 165, 167, 170, 177,
+             191, 194, 204, 209, 216, 218, 223, 228, 231, 261, 263, 266, 268,
+             276, 278, 283, 285, 290, 294, 296, 299, 300, 307, 309, 316, 318,
+             322, 325, 331, 339, 341, 343, 345, 347, 353, 360, 361, 366, 399,
+             400, 401, 408, 410, 412, 415, 418, 420, 422, 425, 427, 431, 433,
+             434, 436, 438, 447, 449, 451, 453, 456, 458, 459, 504, 509, 511,
+             513, 515, 517, 520, 522, 524, 527, 529, 532, 546, 548, 552, 554,
+             557, 559, 562, 564, 566, 568, 570, 572, 580, 585, 590, 592, 595,
+             597, 605, 613, 619, 622, 624, 627, 629, 659, 665, 667, 672, 677,
+             674, 682, 684, 686, 688, 690, 692, 694, 696, 698, 705, 712, 734,
+             742, 744, 747, 749, 751, 753, 755, 757, 759, 767, 769, 819, 827,
+             829, 831, 833, 835, 843, 846, 848, 850, 878],
+        100: [2, 5, 8, 33, 30, 42, 44, 61, 64, 67, 70, 75, 82, 93, 108, 112,
+              114, 117, 122, 125, 126, 148, 153, 156, 159, 176, 180, 188, 190,
+              193, 198, 200, 207, 215, 221, 233, 247, 253, 256, 259, 271, 274,
+              281, 288, 305, 315, 329, 349, 356, 364, 372, 375, 388, 391, 394,
+              397, 404, 430, 444, 478, 496, 499, 502, 507, 536, 541, 544, 575,
+              578, 583, 600, 603, 608, 611, 634, 651, 654, 657, 662, 670, 680,
+              723, 726, 729, 732, 737, 762, 783, 790, 811, 814, 817, 822, 825,
+              838, 857, 860, 886],
+        200: [525, 533, 588, 616, 710],
+        400: [129, 320, 333, 636, 714, 808]
```

Comment:
This seems like data that should go in `data/base_stats.json`, since these are specific to the monster

Either that or in it's own data file; I just don't like having readily-changing data sitting plain in an app

What do you think @billyjbryant ?

Reply:
I thought about the same thing when I realized that there is a data folder with json files few hours ago. Maybe it could be a good idea.

Reply:
I agree, either merged in to base-stats or its own data file in to data folder.

Reply:
I inserted the candy cost of evolutions in `base_stats.json` and modified the evo cost function consequently.